docs: align Rust guide structure with JS (#4)
rticommunity/rticonnextdds-connector-rust: docs/guide/input.md

# Reading data

[`crate::Input`] wraps a DDS `DataReader` and provides an iterator-based
interface over samples returned by the RTI Connector API.

## Getting the input

> 

Reviewer: ```suggestion
To retrieve a specific input from the connector:
```
Or something similar to lead into the code.

```rust
use rtiddsconnector::{Connector, Input};

fn get_input(connector: &Connector) -> rtiddsconnector::ConnectorResult<Input<'_>> {
    connector.get_input("MySubscriber::MyReader")
}
```

## Reading or taking the data

> 

Reviewer: ```suggestion
To make Connext samples available to your Rust application:
```
Not sure if that's accurate, just recommending a lead-in sentence.

  them from the reader.
* [`crate::Input::take`]: removes samples from the reader and places them into
  the local cache.

After calling either, iterate over the cache:

```rust
use rtiddsconnector::Input;

fn process(input: &mut Input) -> rtiddsconnector::ConnectorFallible {
    input.take()?;

    for sample in input.into_iter().valid_only() {
        println!("Sample: {}", sample);
    }

    Ok(())
}
```

`valid_only()` skips samples with invalid data.

## Waiting for data

Use [`crate::Input::wait`] or [`crate::Input::wait_with_timeout`] to block until
data is available. These methods do not read data; call `read` or `take`
afterward.

If you want to wait for data on any input owned by a connector, use
[`crate::Connector::wait_for_data`] or
[`crate::Connector::wait_for_data_with_timeout`]. These methods do not read
samples; call `read` or `take` afterward.

## Accessing data samples

`Sample` provides typed accessors and JSON access:

* [`crate::Sample::get_number`]
* [`crate::Sample::get_boolean`]
* [`crate::Sample::get_string`]
* [`crate::Sample::get_value_json`]

`Sample` implements `Display` to print the full JSON representation of the
sample.

> If you need to access meta-data fields (SampleInfo), see the next section.

Reviewer: That's not the next section; recommend link to Accessing sample meta-data.

## Returning the loan

If you used [`crate::Input::take`], return the loan when you are done using

> [`crate::Input::return_loan`]. This allows the underlying reader to reuse

Reviewer: ```suggestion
[`crate::Input::return_loan`]. This method allows the underlying reader to reuse
```

resources sooner.

> ## Accessing sample meta-data

Reviewer: ```suggestion
## Accessing sample metadata
```

> Every sample includes associated meta-data (SampleInfo). You can access these

Reviewer: ```suggestion
Every sample includes associated metadata (SampleInfo). You can access these
```

Author: we use meta-data in JS

fields using [`crate::Sample::get_info`] and [`crate::Sample::get_info_json`].

> If a sample has invalid data, it still carries meta-data. You can detect this

Reviewer: ```suggestion
If a sample has invalid data, it still carries metadata. You can detect this
```

with [`crate::Sample::is_valid`]. For the list of available info fields and their

> meaning, refer to the RTI Connector documentation.

Reviewer: Add link to the doc.

Author: @Alxe can you enhance the docs for get_info to include the supported fields?

Reviewer 2: I'll do this tomorrow, on this same PR.

Reviewer 2: I've applied these changes in https://github.com/rticommunity/rticonnextdds-connector-rust/pull/4/commits/ed4c211fa3ae7b1075f554eb9a3b75ebac52b494. I'm not particularly happy in adding these, because they'd need to be duplicated in other bindings, but we can approach this issue later on..
